incubator/hnc/e2e/tests/issues_test.go

```diff
@@ -36,4 +42,71 @@ var _ = Describe("Issues", func() {
+		// If the anchor is read, it should unset the 'SubnamespaceAnchorMissing' condition in the subnamespace.
+		// Operation: recreate the 'sub1' subns in 'parent' - kubectl hns create sub1 -n parent
+		// Expected: no conditions.
+		mustRun("kubectl hns create", nsSub1, "-n", nsParent)
+		runShouldNotContain("SubnamespaceAnchorMissing", 1, "kubectl get hierarchyconfigurations.hnc.x-k8s.io -n", nsSub1, "-o yaml")
+	})
+
+	It("Should cascading delete immediate subnamespaces if the anchor is deleted and the subnamespace allows cascadingDelete - issue #501", func() {
+		// set up
+		mustRun("kubectl create ns", nsParent)
+		// Creating the a branch of subnamespace
+		mustRun("kubectl hns create", nsSub1, "-n", nsParent)
+		mustRun("kubectl hns create", nsSub1Sub1, "-n", nsSub1)
+		mustRun("kubectl hns create", nsSub2Sub1, "-n", nsSub1)
+		// If the subnamespace allows cascadingDelete and the anchor is deleted, it should cascading delete all immediate subnamespaces.
+		// Operation: 1) allow cascadingDelete in 'ochid1' - kubectl hns set sub1 --allowCascadingDelete=true
+		// 2) delete 'sub1' subns in 'parent' - kubectl delete subns sub1 -n parent
+		// Expected: 'sub1', 'sub1-sub1', 'sub2-sub1' should all be gone
+		mustRun("kubectl hns set", nsSub1, "--allowCascadingDelete=true")
+		mustRun("kubectl delete subns", nsSub1, "-n", nsParent)
+		runShouldNotContainMultiple([]string {nsSub1, nsSub1Sub1, nsSub2Sub1}, 1, "kubectl hns tree", nsParent)
+	})
+
+	It("SHould cascading delete all subnamespaces if the parent is deleted and allows cascadingDelete - issue #501", func() {
+		// Setting up a 3-level tree with 'parent' as the root
+		mustRun("kubectl create ns", nsParent)
+		// Creating the 1st branch of subnamespace
+		mustRun("kubectl hns create", nsSub1, "-n", nsParent)
+		mustRun("kubectl hns create", nsSub1Sub1, "-n", nsSub1)
+		mustRun("kubectl hns create", nsSub2Sub1, "-n", nsSub1)
+		// Creating the 2nd branch of subnamespaces
+		mustRun("kubectl hns create", nsSub2, "-n", nsParent)
+		mustRun("kubectl hns create", nsSubSub2, "-n", nsSub2)
+		// Creating the 3rd branch of a mix of full and subnamespaces
+		mustRun("kubectl create ns", nsChild)
+		mustRun("kubectl hns set", nsChild, "--parent", nsParent)
+		mustRun("kubectl hns create", nsSubChild, "-n", nsChild)
+		// If the parent namespace allows cascadingDelete and it's deleted, all its subnamespaces should be cascading deleted.
+		// Operation: 1) allow cascadingDelete in 'parent' - kubectl hns set parent --allowCascadingDelete=true
+		// 2) delete 'parent' namespace - kubectl delete ns parent
+		// Expected: only 'fullchild' and 'sub-fullchild' should be left and they should have CRIT_ conditions related to missing 'parent'
+		mustRun("kubectl hns set", nsParent, "--allowCascadingDelete=true")
+		mustRun("kubectl delete ns", nsParent)
+		mustNotRun("kubectl hns tree", nsParent)
+		mustNotRun("kubectl hns tree", nsSub2)
```

Comment:
Since `nsSub1` branch is created earlier, it should be gone too. Add `mustNotRun("kubectl hns tree", nsSub1)`?

Reply:
Done